Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -96,15 +105,44 @@ class BTAnalyticsService: Equatable {
                 return
             }
 
-            let postParameters = self.createAnalyticsEvent(config: configuration, sessionID: self.apiClient.metadata.sessionID, event: event)
-            self.http?.post("v1/tracking/batch/events", parameters: postParameters) { _, _, _ in }
+            if self.timerInterval == 0 {
+                Task {
+                    await self.sendQueuedAnalyticsEvents(configuration: configuration)
+                }
+            } else {
+                self.timer = DispatchSource.makeTimerSource(queue: self.http?.dispatchQueue)
```

Comment:
Currently yes since it reuses the existing connection created via `http.dispatchQueue` - if we want to set up another DispatchQueue for analytics specifically to use a singular queue/timer we can do that. It will add additional latency vs reusing the existing queue/connection based on my local testing. What do you think?

Reply:
I updated this to check if the timer is `nil` before recreating it